pkg/api/components.go

```diff
@@ -270,7 +270,7 @@ func getComponentDefaultContainerImage(component string, cs *ContainerService) s
 	k8sComponents := GetK8sComponentsByVersionMap(kubernetesConfig)[cs.Properties.OrchestratorProfile.OrchestratorVersion]
 	hyperkubeImageBase := kubernetesImageBase
 	hyperkubeImage := hyperkubeImageBase + k8sComponents[common.Hyperkube]
-	if cs.Properties.IsAzureStackCloud() {
+	if cs.Properties.IsAzureStackCloud() && !common.IsKubernetesVersionGe(cs.Properties.OrchestratorProfile.OrchestratorVersion, "1.21.0") {
```

Comment:
include please a small comment explaining why the behavior changes starting with 1.21

Reply:
the alternative is to write code that document itself

```
variableWithMeaningfulName := !common.IsKubernetesVersionGe(...)
if cs.Properties.IsAzureStackCloud() && variableWithMeaningfulName { ... }
```

Reply:
added comment